src/plugins/controls/public/control_group/editor/control_editor.tsx

```diff
@@ -301,9 +301,7 @@ export const ControlEditor = ({
                   const newDefaultTitle = field.displayName ?? field.name;
                   setDefaultTitle(newDefaultTitle);
                   setSelectedField(field.name);
-                  setSelectedControlType(
-                    fieldRegistry?.[field.displayName].compatibleControlTypes[0]
-                  );
+                  setSelectedControlType(fieldRegistry?.[field.name].compatibleControlTypes[0]);
```

Comment:
Should this be updated to something that handles case where field.name does not exist `fieldRegistry?.[field.name]?.compatibleControlTypes[0]`

Reply:
Good call - good to be extra safe I think. Done in 03a7f78ac715246a3e899be74d97273ba76a9a6a